synthesizer/src/vm/execute.rs

```diff
@@ -224,7 +234,11 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
                 lap!(timer, "Execute the call");
 
                 // Prepare the assignments.
-                cast_mut_ref!(trace as Trace<N>).prepare(query)?;
+                if let Some(query) = query {
```

Comment:
It would be a little nicer to use similar code to what you removed above:

```suggestion
let query = query.unwrap_or_else(|| Query::VM(self.block_store().clone());
cast_mut_ref!(trace as Trace<N>).prepare(query)?;
```

I am unfamiliar with the metrics code, but is it fine to remove `lap!(timer, "Prepare the query");` from this function?

Reply:
The `unwrap_or_else` can't be used due to the clash between generic and concrete types. Trait objects can't be used either, as the trait is not object safe. The alternative would be a dedicated `enum`, but it would result in a greater diff.

I don't use the metrics stuff either, but it doesn't seem like it's meaningful, as the creation of the query takes an insignificant amount of time.

Reply:
Oh, okay. That makes sense.